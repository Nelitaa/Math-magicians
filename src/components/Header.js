import React from 'react';
import { Link } from 'react-router-dom';

function Header() {
  return (
    <header className="background-style">
      <nav>
        <h1>Math Magicians</h1>
        <button type="button" id="mobile-nav" className="background-style">
          <span />
          <span />
          <span />
        </button>
        <ul id="desktop-nav">
          <li><Link to="/">Home</Link></li>
          <li><Link to="/calculator">Calculator</Link></li>
          <li><Link to="/quote">Quote</Link></li>
        </ul>
      </nav>
    </header>
  );
}

export default Header;
