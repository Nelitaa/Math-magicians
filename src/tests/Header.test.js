import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import Header from '../components/Header';

test('renders Header component', () => {
  const { asFragment } = render(<Router><Header /></Router>);
  expect(asFragment()).toMatchSnapshot();
});

test('renders Header component with links', () => {
  const { getByText } = render(<Router><Header /></Router>);
  expect(getByText('Home')).toHaveAttribute('href', '/');
  expect(getByText('Calculator')).toHaveAttribute('href', '/calculator');
  expect(getByText('Quote')).toHaveAttribute('href', '/quote');
});
