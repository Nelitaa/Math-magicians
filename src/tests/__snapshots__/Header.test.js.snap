// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header component 1`] = `
<DocumentFragment>
  <header
    class="background-style"
  >
    <nav>
      <h1>
        Math Magicians
      </h1>
      <button
        class="background-style"
        id="mobile-nav"
        type="button"
      >
        <span />
        <span />
        <span />
      </button>
      <ul
        id="desktop-nav"
      >
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            href="/quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
